lib/domain/platform/i_network_info.dart

```diff
@@ -0,0 +1,3 @@
+abstract class INetworkInfo {
+  Future<bool> get isConnected;
+}
```

Comment:
suggestion: one of the best parts of knowing if the user has a network connection is to inform them that they are experiencing errors because of the lack of connection.  I think it is great this was used proactively to prevent crashes though.  I think informing the user of this would be slightly better.

Reply:
Thanks! I improved the Failure handling to inform the user about the different errors. See (https://github.com/Superformula/mobile-test/pull/53/commits/b356b56216404f449d27b545d21355ed6bce5fad) and (https://github.com/Superformula/mobile-test/pull/53/commits/1bb58ede7429ce77a377c6bf25a6aa0cfb5bd60e)